### ISSUE TITLE
Game doesn't add and subtract final letters when someone goes out

### ISSUE THREAD

**Issue by pkolano** [cdot/Xanado#24]
Just played first full game and noticed that the final letters in everyone's hand are not added to the winners and and subtracted from the loser's as they're supposed to be.  We challenged (that failed) the last word so don't know if that had something to do with it.  Thanks.

**Comment by pkolano** [comment]
Actually, now that I look closer, the game should be over, but it isn't.  It let the player after the one that went out still go.  Then, noone else could play and then it was impossible to end it.

**Comment by pkolano** [comment]
But then it reached a state on the main screen when the game was over after the person who challenged the last word finally accepted the word the second time around.

**Comment by pkolano** [comment]
Also, the final scores did not reflect letters being added and subtracted.  It seemed like one person's letter were subtracted, but were not added to the winner and noone else's scrores were affected.

**Comment by cdot** [comment]
That's been a source of problems in the past, but I thought I had fixed it. It seems to work when I try it, but there may be edge cases. If you can provide any more detail, that might help to build a test case.

**Comment by pkolano** [comment]
It wasn't any special scenario.  4 player game.  One person went out and the next person challenged the last word (as a sidenote, I was wondering if the last word should always be challenged automatically since there is no penalty for a failed challenge at that point).  The words were good, but the game didn't end.  Instead, it allowed the next person to go.  In our latest game, there was an exception generated:

<-- 500  Error: Found more than one player with no tiles when finishing game
    at /home/nodejs/Xanado-master/js/game/Game.js:1052:13          at Array.forEach (<anonymous>)
    at Game.confirmGameOver (/home/nodejs/Xanado-master/js/game/Game.js:1048:17)
    at /home/nodejs/Xanado-master/js/server/Server.js:850:21

The only other interesting thing I see is below.  Maybe the problem is that you are setting "nextToGoKey" to something when actually nobody should be next to go since the game is over?  

<-S- * turn Turn {
  words: [],
  score: 0,
  bonus: 0,
  placements: [],
  replacements: [],
  type: 'challenge-failed',
  gameKey: '334d3b26522ef36f',
  playerKey: '32dv07ptvr',
  nextToGoKey: '4k2ckkcfs9',
  emptyPlayerKey: '36ztocgyjd',
  challengerKey: undefined,
  timestamp: 1649395807768
}

I've only played two games that were both challenged at the end so the bad scoring could just be due to the final challenge bug, but my family said when they played a different time, the final words weren't challenged and the scoring still didn't seem to be correct, but I didn't witness it myself.
